rotkehlchen/chain/evm/decoding/rainbow/decoder.py

```diff
@@ -0,0 +1,402 @@
+        token and input asset is eth
+        - EVM token to EVM token swap, looking for input and output assets are tokens
+        It attempts to find the necessary logs for wrapping and unwrapping tokens
+        and decodes the swap accordingly.
+        If the required logs are missing, it logs an error and returns
+        the original decoded events.
+        """
+        sender = transaction.from_address
+        in_event, out_event = self._get_in_out_events(decoded_events, sender)
+        if in_event is None or out_event is None:
+            log.error(
+                f'Failed to post decoding rainbow {transaction} '
+                f'due to missing in_event or out_event.',
+            )
+            return decoded_events
+
+        out_asset_is_evm_token = out_event.asset.is_evm_token()
+        in_asset_is_evm_token = in_event.asset.is_evm_token()
+
+        if out_event.asset == self.evm_inquirer.native_token and in_asset_is_evm_token:
+            # Try to find if have wrap eth into weth log
+            if (wrap_event := self._find_event_by_log(all_logs, WETH_DEPOSIT_TOPIC)) is None:
+                log.error(
+                    f'Failed to decode fee from swap involving {self.evm_inquirer.native_token} '
+                    f'to token for transaction {transaction} due to missing wrap log.',
+                )
+                return decoded_events
+
+            self._maybe_decode_fee_from_swap_native_to_token(
+                transaction=transaction,
+                out_event=out_event,
+                in_event=in_event,
+                decoded_events=decoded_events,
+                sender=sender,
+                wrap_event=wrap_event,
+            )
+        elif out_asset_is_evm_token and in_event.asset == self.evm_inquirer.native_token:
+            if (unwrap_event := self._find_event_by_log(all_logs, WETH_WITHDRAW_TOPIC)) is None:
+                log.error(
+                    f'Failed to decode fee from swap involving {self.evm_inquirer.native_token} '
+                    f'to token for transaction {transaction} due to missing unwrap log.',
+                )
+                return decoded_events
+
+            self._maybe_decode_fee_from_swap_token_to_native(
+                transaction=transaction,
+                decoded_events=decoded_events,
+                in_event=in_event,
+                out_event=out_event,
+                sender=sender,
+                unwrap_event=unwrap_event,
+            )
+        elif out_asset_is_evm_token and in_asset_is_evm_token:
+            self._maybe_decode_fee_from_swap_token_to_token(
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                sender=sender,
+                out_event=out_event,
+                in_event=in_event,
+            )
+
+        return decoded_events
+
+    def _maybe_enricher_rainbow_transfer(
```

Comment:
```suggestion
    def _maybe_enrich_rainbow_transfer(
```
Should use the verb `enrich` here instead of the noun `enricher`.